src/CodeGen_C.cpp

```diff
@@ -2458,6 +2476,110 @@ void CodeGen_C::visit(const Call *op) {
             }
             rhs << "(&" << struct_name << ")";
         }
+    } else if (op->is_intrinsic(Call::declare_struct_type)) {
+        // Emit a declaration like:
+        // struct {const int f_0, const char f_1, const int f_2} foo;
+
+        // Declares a struct type. Returns a null pointer of the new type.
+        internal_assert(op->args.size() >= 2);
+
+        const StringImm *str_imm = op->args[0].as<StringImm>();
+        internal_assert(str_imm != nullptr);
+        std::string name = c_print_name(str_imm->value, false);
+
+        const int64_t *mode = as_const_int(op->args[1]);
+        internal_assert(mode != nullptr);
+
+        if (*mode == 1 || !starts_with(name, "halide_")) {
```

Comment:
The meaning of `mode` really should be documented somewhere, and the legal values probably deserve named constants. Having to read the implementation to figure out what it's for is not ideal.